homeassistant/components/device_tracker/__init__.py

```diff
@@ -136,75 +111,52 @@ def see(hass: HomeAssistantType, mac: str = None, dev_id: str = None,
 
 async def async_setup(hass: HomeAssistantType, config: ConfigType):
     """Set up the device tracker."""
-    yaml_path = hass.config.path(YAML_DEVICES)
+    tracker = await legacy.get_tracker(hass, config)
 
-    conf = config.get(DOMAIN, [])
-    conf = conf[0] if conf else {}
-    consider_home = conf.get(CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME)
+    async def setup_entry_helper(entry):
+        """Set up a config entry."""
+        platform = await setup.async_create_platform_type(
+            hass, config, entry.domain, entry)
 
-    defaults = conf.get(CONF_NEW_DEVICE_DEFAULTS, {})
-    track_new = conf.get(CONF_TRACK_NEW)
-    if track_new is None:
-        track_new = defaults.get(CONF_TRACK_NEW, DEFAULT_TRACK_NEW)
+        if platform is None:
+            return False
 
-    devices = await async_load_config(yaml_path, hass, consider_home)
-    tracker = DeviceTracker(
-        hass, consider_home, track_new, defaults, devices)
+        await platform.async_setup_legacy(hass, tracker)
 
-    async def async_setup_platform(p_type, p_config, disc_info=None):
-        """Set up a device tracker platform."""
-        platform = await async_prepare_setup_platform(
-            hass, config, DOMAIN, p_type)
-        if platform is None:
-            return
+        return True
+
+    hass.data[DOMAIN] = setup_entry_helper
```

Comment:
We still won't support entry unloading with this approach. Do you have a plan for that?

Reply:
The plan is that all config entry platforms will be required to use EntityPlatform, and we will forward calls for setup and unload to the entity component to it's work as usual.

Reply:
But that's going to be part of a future PR.

Reply:
Do you mean we will extend `setup_entry_helper` to support entry unloading?

Reply:
Compare with standard component:
https://github.com/home-assistant/home-assistant/blob/94a2fd542ea47abf59d2e3a21f17637a3d9af92f/homeassistant/components/sensor/__init__.py#L46-L53

Reply:
I expect to adopt the standard component approach.

Reply:
I don't see how we can support entry unloading with the standard approach during the transition period at the same time as keeping the `setup_entry_helper` in hass.data under the domain.

Reply:
The setup_entry_helper will go away. We will need to migrate all those platforms that use config entries to use an entity. Which we would adopt from your PR .

Reply:
Right. Thanks for the clarification! I was confused.